diff --git a/modules/host_cpu_info.pm b/modules/host_cpu_info.pm
index f90453a..bd3c4dc 100644
--- a/modules/host_cpu_info.pm
+++ b/modules/host_cpu_info.pm
@@ -125,7 +125,7 @@ sub host_cpu_info
              }
           else
              {
-             $value = simplify_number(convert_number($$values[0][2]->value) * 0.01);
+             $value = simplify_number(convert_number($$values[0][3]->value) * 0.01);
              }
 
           if ($subselect eq "all")
