diff --git a/graphite_api/config.py b/graphite_api/config.py
index 5e1e382..530e568 100644
--- a/graphite_api/config.py
+++ b/graphite_api/config.py
@@ -42,7 +42,7 @@ default_conf = {
             '/srv/graphite/whisper',
         ],
     },
-    'time_zone': get_localzone().zone,
+    'time_zone': 'UTC',
 }
 if default_conf['time_zone'] == 'local':  # tzlocal didn't find anything
     default_conf['time_zone'] = 'UTC'
